src/main/java/org/eclipse/persistence/platform/database/SQLServerPlatform.java

```diff
@@ -699,7 +699,9 @@ public boolean supportsIdentity() {
      */
     @Override
     public boolean supportsSequenceObjects() {
-        return isVersion11OrHigher;
+        // PATCHED Without this EclipseLink handles SEQUENCE field from old databases wrongly.
+        // See https://bugs.eclipse.org/bugs/show_bug.cgi?id=499233
+        return false;
```

Comment:
Is there no more precise way of telling when to use `SEQUENCE` and when not to? I'm no database expert but I think it would be a performance impact to always use `IDENTITY`.

Reply:
There is a better way, but it'll most likely require some deeper plumping inside EclipseLink. I think for somebody not too familiar with EclipseLink's intrinsics it will pose quite some work.

The issue basically appears because the existing table and the table EclipseLink expects have a mismatch. This patch brings EclipseLink back to its old expectation, which it already had some versions ago and which matches the existing tables.

We already had this same patch applied before, but it's gone missing during the upgrade to 2.7.4.